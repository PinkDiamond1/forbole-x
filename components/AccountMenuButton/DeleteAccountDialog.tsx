import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  IconButton,
  Typography,
} from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import CloseIcon from '../../assets/images/icons/icon_cross.svg'
import useStyles from './styles'
import { useWalletsContext } from '../../contexts/WalletsContext'
import useIconProps from '../../misc/useIconProps'

interface DeleteAccountDialogProps {
  accountAddress: string
  open: boolean
  onClose(): void
}

const DeleteAccountDialog: React.FC<DeleteAccountDialogProps> = ({
  accountAddress,
  open,
  onClose,
}) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const iconProps = useIconProps()
  const { deleteAccount } = useWalletsContext()

  const onButtonClick = React.useCallback(async () => {
    try {
      await deleteAccount(accountAddress)
    } catch (err) {
      console.log(err)
    }
  }, [deleteAccount, accountAddress])

  return (
    <Dialog open={open} onClose={onClose}>
      <IconButton className={classes.closeButton} onClick={onClose}>
        <CloseIcon {...iconProps} />
      </IconButton>
      <DialogTitle>{t('delete account')}</DialogTitle>
      <DialogContent>
        <Typography>{t('delete account warning')}</Typography>
      </DialogContent>
      <DialogActions>
        <Button
          className={classes.dialogButton}
          variant="contained"
          color="secondary"
          onClick={onClose}
        >
          {t('cancel')}
        </Button>
        <Button
          className={classes.dialogButton}
          variant="contained"
          color="primary"
          onClick={onButtonClick}
        >
          {t('delete')}
        </Button>
      </DialogActions>
    </Dialog>
  )
}

export default DeleteAccountDialog
