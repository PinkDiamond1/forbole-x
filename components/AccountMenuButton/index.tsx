import { Button, IconButton, Menu, MenuItem, Box } from '@material-ui/core'
import React from 'react'
import useTranslation from 'next-translate/useTranslation'
import useStyles from './styles'
import MoreIcon from '../../assets/images/icons/icon_more.svg'
import useIconProps from '../../misc/useIconProps'
import ChangeAccountMonikerDialog from './ChangeAccountMonikerDialog'
import DeleteAccountDialog from './DeleteAccountDialog'

const AccountMenuButton: React.FC<{
  account: Account
  buttonComponent?: React.ReactElement
}> = ({ account, buttonComponent }) => {
  const { t } = useTranslation('common')
  const iconProps = useIconProps()
  const classes = useStyles()
  const [anchor, setAnchor] = React.useState<Element>()

  const [changeAccountNameOpen, setChangeAccountNameOpen] = React.useState(false)
  const [deleteAccountOpen, setDeleteAccountOpen] = React.useState(false)

  const onClose = React.useCallback(() => setAnchor(undefined), [setAnchor])

  return (
    <>
      {buttonComponent ? (
        React.cloneElement(buttonComponent, { onClick: (e) => setAnchor(e.currentTarget) })
      ) : (
        <IconButton onClick={(e) => setAnchor(e.currentTarget)}>
          <MoreIcon {...iconProps} />
        </IconButton>
      )}
      <Menu
        anchorEl={anchor}
        getContentAnchorEl={null}
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'center',
        }}
        transformOrigin={{
          vertical: 'top',
          horizontal: 'center',
        }}
        keepMounted
        open={!!anchor}
        onClose={onClose}
      >
        <MenuItem
          className={classes.menuItem}
          button
          onClick={() => {
            setChangeAccountNameOpen(true)
            onClose()
          }}
        >
          {t('change account moniker')}
        </MenuItem>
        <Box m={2}>
          <Button
            fullWidth
            variant="contained"
            color="primary"
            onClick={() => {
              setDeleteAccountOpen(true)
              onClose()
            }}
          >
            {t('delete account')}
          </Button>
        </Box>
      </Menu>
      <ChangeAccountMonikerDialog
        open={changeAccountNameOpen}
        onClose={() => setChangeAccountNameOpen(false)}
        account={account}
      />
      <DeleteAccountDialog
        open={deleteAccountOpen}
        onClose={() => setDeleteAccountOpen(false)}
        account={account}
      />
    </>
  )
}

export default React.memo(AccountMenuButton)
