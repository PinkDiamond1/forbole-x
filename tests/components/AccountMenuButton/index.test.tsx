import React from 'react'
import renderer from 'react-test-renderer'
import AccountMenuButton from '../../../components/AccountMenuButton'

jest.mock('@material-ui/core/Menu', () => (props) => <div id="Menu" {...props} />)
jest.mock('../../../components/AccountMenuButton/ChangeAccountMonikerDialog', () => (props) => (
  <div id="ChangeAccountMonikerDialog" {...props} />
))
jest.mock('../../../components/AccountMenuButton/DeleteAccountDialog', () => (props) => (
  <div id="DeleteAccountDialog" {...props} />
))

describe('component: AccountMenuButton', () => {
  it('renders correctly', () => {
    const component = renderer.create(<AccountMenuButton accountAddress="123" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders open / close menu correctly', () => {
    const component = renderer.create(<AccountMenuButton accountAddress="123" />)
    renderer.act(() => {
      component.root.findAllByType('button')[0].props.onClick({ currentTarget: 'anchor' })
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    renderer.act(() => {
      component.root.findByProps({ id: 'Menu' }).props.onClose()
    })
    const closedTree = component.toJSON()
    expect(closedTree).toMatchSnapshot()
  })
  it('renders open / close change account name dialog correctly', () => {
    const component = renderer.create(<AccountMenuButton accountAddress="123" />)
    renderer.act(() => {
      component.root.findAllByType('button')[0].props.onClick({ currentTarget: 'anchor' })
    })
    renderer.act(() => {
      component.root.findAllByType('li')[0].props.onClick()
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    renderer.act(() => {
      component.root.findByProps({ id: 'ChangeAccountMonikerDialog' }).props.onClose()
    })
    const closedTree = component.toJSON()
    expect(closedTree).toMatchSnapshot()
  })
  it('renders open / close delete wallet dialog correctly', () => {
    const component = renderer.create(<AccountMenuButton accountAddress="123" />)
    renderer.act(() => {
      component.root.findAllByType('button')[0].props.onClick({ currentTarget: 'anchor' })
    })
    renderer.act(() => {
      component.root.findAllByType('button')[1].props.onClick()
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    renderer.act(() => {
      component.root.findByProps({ id: 'DeleteAccountDialog' }).props.onClose()
    })
    const closedTree = component.toJSON()
    expect(closedTree).toMatchSnapshot()
  })
})

afterEach(() => {
  jest.clearAllMocks()
})
