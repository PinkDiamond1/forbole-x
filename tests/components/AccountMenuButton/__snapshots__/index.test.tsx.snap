// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: AccountMenuButton renders correctly 1`] = `
Array [
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_more"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
  </button>,
  <div
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    getContentAnchorEl={null}
    id="Menu"
    keepMounted={true}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root HookGlobalStyles-menuItem-1 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      common:change account moniker
    </li>
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:delete account
        </span>
      </button>
    </div>
  </div>,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="ChangeAccountMonikerDialog"
    onClose={[Function]}
    open={false}
  />,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="DeleteAccountDialog"
    onClose={[Function]}
    open={false}
  />,
]
`;

exports[`component: AccountMenuButton renders open / close change account name dialog correctly 1`] = `
Array [
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_more"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <div
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    getContentAnchorEl={null}
    id="Menu"
    keepMounted={true}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root HookGlobalStyles-menuItem-1 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      common:change account moniker
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <div
      className="MuiBox-root MuiBox-root-11"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:delete account
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="ChangeAccountMonikerDialog"
    onClose={[Function]}
    open={true}
  />,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="DeleteAccountDialog"
    onClose={[Function]}
    open={false}
  />,
]
`;

exports[`component: AccountMenuButton renders open / close change account name dialog correctly 2`] = `
Array [
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_more"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <div
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    getContentAnchorEl={null}
    id="Menu"
    keepMounted={true}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root HookGlobalStyles-menuItem-1 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      common:change account moniker
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <div
      className="MuiBox-root MuiBox-root-11"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:delete account
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="ChangeAccountMonikerDialog"
    onClose={[Function]}
    open={false}
  />,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="DeleteAccountDialog"
    onClose={[Function]}
    open={false}
  />,
]
`;

exports[`component: AccountMenuButton renders open / close delete wallet dialog correctly 1`] = `
Array [
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_more"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <div
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    getContentAnchorEl={null}
    id="Menu"
    keepMounted={true}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root HookGlobalStyles-menuItem-1 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      common:change account moniker
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <div
      className="MuiBox-root MuiBox-root-12"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:delete account
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="ChangeAccountMonikerDialog"
    onClose={[Function]}
    open={false}
  />,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="DeleteAccountDialog"
    onClose={[Function]}
    open={true}
  />,
]
`;

exports[`component: AccountMenuButton renders open / close delete wallet dialog correctly 2`] = `
Array [
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_more"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <div
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    getContentAnchorEl={null}
    id="Menu"
    keepMounted={true}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root HookGlobalStyles-menuItem-1 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      common:change account moniker
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <div
      className="MuiBox-root MuiBox-root-12"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:delete account
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="ChangeAccountMonikerDialog"
    onClose={[Function]}
    open={false}
  />,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="DeleteAccountDialog"
    onClose={[Function]}
    open={false}
  />,
]
`;

exports[`component: AccountMenuButton renders open / close menu correctly 1`] = `
Array [
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_more"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <div
    anchorEl="anchor"
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    getContentAnchorEl={null}
    id="Menu"
    keepMounted={true}
    onClose={[Function]}
    open={true}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root HookGlobalStyles-menuItem-1 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      common:change account moniker
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:delete account
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="ChangeAccountMonikerDialog"
    onClose={[Function]}
    open={false}
  />,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="DeleteAccountDialog"
    onClose={[Function]}
    open={false}
  />,
]
`;

exports[`component: AccountMenuButton renders open / close menu correctly 2`] = `
Array [
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        data-file-name="SvgIcon_more"
        fill="#e0e0e0"
        height={16}
        width={16}
      />
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>,
  <div
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    getContentAnchorEl={null}
    id="Menu"
    keepMounted={true}
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <li
      aria-disabled={false}
      className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root HookGlobalStyles-menuItem-1 MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="menuitem"
      tabIndex={-1}
    >
      common:change account moniker
      <span
        className="MuiTouchRipple-root"
      />
    </li>
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:delete account
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="ChangeAccountMonikerDialog"
    onClose={[Function]}
    open={false}
  />,
  <div
    account={
      Object {
        "address": "address",
        "createdAt": 0,
        "crypto": "DSM",
        "fav": false,
        "index": 0,
        "name": "name",
        "walletId": "123",
      }
    }
    id="DeleteAccountDialog"
    onClose={[Function]}
    open={false}
  />,
]
`;
